lightning/src/ln/peer_handler.rs

```diff
@@ -720,8 +720,8 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref> PeerManager<D
 													log_debug!(self.logger, "Error handling message; disconnecting peer with: {}", e.err);
 													return Err(PeerHandleError{ no_connection_possible: false });
 												},
-												msgs::ErrorAction::IgnoreCommonError => {
-													log_trace!(self.logger, "Error handling message; ignoring: {}", e.err);
+												msgs::ErrorAction::IgnoreAndLog(level) => {
+													log_given_level!(self.logger, cmp::min(Level::Debug, level), "Error handling message; ignoring: {}", e.err);
```

Comment:
Why is the `cmp::min` used? Would it make sense to write a custom macro for this so it doesn't need to repeated elsewhere?

Reply:
Hmm, I thought the "minimum level" there was pretty readable. I could move the `cmp::min` into the macro, but it felt readable and this is slightly more generic because it would let you do some match or if statement that sets the level?

Reply:
To rephrase my first question, why not just use `level` here? I don't quite understand why we want to use a lower level if `IgnoreAndLog` gives, say, `Warn`.

> Hmm, I thought the "minimum level" there was pretty readable. I could move the `cmp::min` into the macro, but it felt readable and this is slightly more generic because it would let you do some match or if statement that sets the level?

I was saying to leave `log_given_level` as is and add another macro like `log_debug_or_lower` (or similarly named) to avoid a lot of boilerplate if this pattern will be common.

Reply:
> To rephrase my first question, why not just use level here? I don't quite understand why we want to use a lower level if IgnoreAndLog gives, say, Warn.

Hmm, right, I guess we could? Its a bit strange to "warn" the user for an error that we "ignored", but I can just drop it.

> I was saying to leave log_given_level as is and add another macro like log_debug_or_lower (or similarly named) to avoid a lot of boilerplate if this pattern will be common.

I mean we only handle the `ErrorAction` stuff in two places... :)